version/version.proto

```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+
+package version;
+
+import "google/protobuf/descriptor.proto";
+
+/* Application-level versioning shall use a 3-part version number (M.m.p) following semver rules */
+/* 1. MAJOR version (M): increment when you make incompatible changes                            */
+/* 2. MINOR version (m): increment when you add functionality in backward compatible manner      */
+/* 3. PATCH version (p): increment when you make backward compatible bug fixes                   */
+
+// Extensions of file-level options for interface version: should *not* be modified
+extend google.protobuf.FileOptions {
+  uint32 interface_major_version = 50001;
```

Comment:
Does 50000 have special meaning here?

Reply:
Yes, it's the start of the [range recommended for custom options](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/descriptor.proto#L327) in gRPC